rasa/core/actions/action.py

```diff
@@ -940,3 +949,141 @@ def name(self) -> Text:
     def __init__(self) -> None:
         """Initializes action default ask rephrase."""
         super().__init__("utter_ask_rephrase", silent_fail=True)
+
+
+class ActionExtractSlots(Action):
+    """Silent action that runs after each user turn.
+
+    Sets slots to extracted values from user message
+    according to assigned slot mappings.
+    """
+
+    def __init__(self, action_endpoint: Optional[EndpointConfig]) -> None:
+        """Initializes default action extract slots."""
+        self.action_endpoint = action_endpoint
+
+    def name(self) -> Text:
+        """Returns action_extract_slots name."""
+        return ACTION_EXTRACT_SLOTS
+
+    async def run(
+        self,
+        output_channel: "OutputChannel",
+        nlg: "NaturalLanguageGenerator",
+        tracker: "DialogueStateTracker",
+        domain: "Domain",
+    ) -> List[Event]:
+        """Runs action. Please see parent class for the full docstring."""
+        slot_events: List[Event] = []
+
+        default_slots = {
+            rasa.shared.core.constants.REQUESTED_SLOT,
+            rasa.shared.core.constants.SESSION_START_METADATA_SLOT,
+            rasa.shared.core.constants.SLOT_LISTED_ITEMS,
+            rasa.shared.core.constants.SLOT_LAST_OBJECT,
+            rasa.shared.core.constants.SLOT_LAST_OBJECT_TYPE,
+        }
```

Comment:
`rasa.shared.core.constants` already contains `DEFAULT_ACTION_NAMES` — can we do the same with slot names? Or you think it's better to keep them here?

Asking because my opinion is that this method (`run`) is quite a long one, and I would be happy if we restructure it a bit.